node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -110,21 +112,18 @@ public void converge(State wantedState) {
             adjustNodeAgentsToRunFromNodeRepository();
             nodeAdmin.setFrozen(false);
 
-            NodeState currentNodeState = nodeRepository.getNode(hostHostname).state();
-            if (currentNodeState == NodeState.active) orchestrator.resume(hostHostname);
+            if (hostIsActiveInNR) orchestrator.resume(hostHostname);
 
             throw new ConvergenceException("Timed out trying to freeze all nodes: will force an unfrozen tick");
         }
 
-        if (currentState == wantedState) return;
+        boolean wantFrozen = wantedState != RESUMED;
+        if (currentState == wantedState && wantFrozen == node.orchestratorStatus().isSuspended()) return;
```

Comment:
This will return even if wantedState is SUSPENDED, the host is suspended, and a child is resumed.  Which is one of the use-case you're trying to fix?

Case B: If the wanted state is RESUMED and !hostIsActiveInNR, then I believe isSuspended() is true since it is UNKNOWN and != NO_REMARKS.  As it is now it would mean printing "State changed from ... to ..." every tick.  Instead we'd like to return here.

Reply:
The main use-case is when HA converges before switching to real, typically it is resuming after updating HA. After it switches, it wont resume against real and the upgrade halts.

Case B: As far as I can see from the code, `orchestratorStatus` is not set in node response if the node is not allocated, that is also the case for `NO_REMARKS`, so it should be `NO_REMARKS` in `NodeSpec::orchestratorStatus`?

Reply:
Right, but you also mentioned operators: So if an operator forces resume of a child.  Not a big deal.  But the logic is a bit harder to follow since it doesn't match the below.

Case B: Agree.